br/pkg/lightning/restore/get_pre_info.go

```diff
@@ -0,0 +1,746 @@
+}
+
+// TargetInfoGetter defines the operations to get information from target.
+type TargetInfoGetter interface {
+	// FetchRemoteTableModels fetches the table structures from the remote target.
+	FetchRemoteTableModels(ctx context.Context, schemaName string) ([]*model.TableInfo, error)
+	// CheckVersionRequirements performs the check whether the target satisfies the version requirements.
+	CheckVersionRequirements(ctx context.Context) error
+	// DoesTableContainData checks whether the specified table on the target DB contains data or not.
+	DoesTableContainData(ctx context.Context, schemaName string, tableName string) (bool, error)
+	// GetTargetSysVariablesForImport gets some important systam variables for importing on the target.
+	GetTargetSysVariablesForImport(ctx context.Context) map[string]string
+	// GetReplicationConfig gets the replication config on the target.
+	GetReplicationConfig(ctx context.Context) (*pdtypes.ReplicationConfig, error)
+	// GetStorageInfo gets the storage information on the target.
+	GetStorageInfo(ctx context.Context) (*pdtypes.StoresInfo, error)
+	// GetEmptyRegionsInfo gets the region information of all the empty regions on the target.
+	GetEmptyRegionsInfo(ctx context.Context) (*pdtypes.RegionsInfo, error)
+}
+
+type preInfoGetterKey string
+
+const (
+	preInfoGetterKeyDBMetas preInfoGetterKey = "PRE_INFO_GETTER/DB_METAS"
+)
+
+func withPreInfoGetterKeyValue(ctx context.Context, key preInfoGetterKey, val any) context.Context {
+	return context.WithValue(ctx, key, val)
+}
+
+// TargetInfoGetterImpl implements the operations to get information from the target.
+type TargetInfoGetterImpl struct {
+	cfg          *config.Config
+	targetDBGlue glue.Glue
+	tls          *common.TLS
+	backend      backend.TargetInfoGetter
+}
+
+// NewTargetInfoGetterImpl creates a TargetInfoGetterImpl object.
+func NewTargetInfoGetterImpl(
+	cfg *config.Config,
+	targetDB *sql.DB,
+) (*TargetInfoGetterImpl, error) {
+	targetDBGlue := glue.NewExternalTiDBGlue(targetDB, cfg.TiDB.SQLMode)
+	tls, err := cfg.ToTLS()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	var backendTargetInfoGetter backend.TargetInfoGetter
+	switch cfg.TikvImporter.Backend {
+	case config.BackendTiDB:
+		backendTargetInfoGetter = tidb.NewTargetInfoGetter(targetDB)
+	case config.BackendLocal:
+		backendTargetInfoGetter = local.NewTargetInfoGetter(tls, targetDBGlue, cfg.TiDB.PdAddr)
+	default:
+		return nil, common.ErrUnknownBackend.GenWithStackByArgs(cfg.TikvImporter.Backend)
+	}
+	return &TargetInfoGetterImpl{
+		targetDBGlue: targetDBGlue,
+		tls:          tls,
+		backend:      backendTargetInfoGetter,
+	}, nil
+}
+
+// FetchRemoteTableModels fetches the table structures from the remote target.
+// It implements the TargetInfoGetter interface.
+func (g *TargetInfoGetterImpl) FetchRemoteTableModels(ctx context.Context, schemaName string) ([]*model.TableInfo, error) {
+	if !g.targetDBGlue.OwnsSQLExecutor() {
```

Comment:
When `g.targetDBGlue` not `OwnsSQLExecutor` then `GetTables` from it? Is it typo?

Reply:
Does it make any sense? cc @gozssky

Reply:
`g.targetDBGlue.OwnsSQLExecutor` is used to check whether it's a mocked glue, mainly used for test